### DISCUSSION TITLE
Start up apps inteferring with browser launch

### DISCUSSION THREAD

**Discussion by VirtualRusty** [SafeExamBrowser/seb-win-refactoring#547] [Q&A]
We run exams via BYO devices and a lot of students have apps running at laptop startup that interfere with the launch of SEB, things like antivirus software, Steam and TeamViewer (or its daemon). Shouldn't SEB be able to launch without these apps stopping it? We don't have the time and/or skills to deal with these issues with a 200+ seat room. I thought at least Steam and TeamViewer should be quite common launch environments encountered, based on the number of students we find running these.

**Comment by danschlet** [comment]
In general, SEB needs to terminate apps which could be used to cheat in exams. As you might imagine, TeamViewer or other apps which have screen sharing features (I guess that applies to Steam as well) can be used to transmit screen content to other people outside of an exam and with TeamViewer, someone outside of an exam hall could even solve an exam for a student.

Unfortunately Windows doesn't offer a built-in kiosk (lockdown/single app) mode for BYOD scenarios. Therefore we need to use various techniques to stop apps which could be used for cheating while SEB is running. In some cases, the only possible way is to terminate such apps while SEB is running. SEB has the capability to terminate most apps while starting up automatically, without prompting the user. In [Applications/Prohibited Processes](https://safeexambrowser.org/windows/win_usermanual_en.html#ProhibitedProcessesSection) you have to set the "Force quit" property for each app you want to terminate automatically. Just be aware that if that applications has some documents open, users might loose those. Probably that's not really an issue with TeamViewer and Steam though, so "Force quit" could be set.

You didn't mention that, but does terminating these apps when SEB asks for it while starting up not work? Do they need to be deactivated manually or even uninstalled? It's possible that SEB cannot terminate apps which are running with administrator rights.

Regarding anti malware software we always try to report incompatibilities to their vendors and often they would add SEB to a list of exceptions to avoid false alarms. You can also report which anti virus software causes an issue and we will try to handle that.

To mitigate the problem, you could create a list of common apps which SEB wants to terminate and communicate to students before the test that they should make sure these apps are not running or are deactivated. 

We are aware that this isn't ideal, but as long as Microsoft doesn't finally understand that they need to add an assessment/test taking/kiosk mode to Windows which can be used with BYOD, there will always be a need to stop some apps from recording or transmitting exam content. Apple for example has added such BYOD assessment modes to both iOS and macOS, so we hope Microsoft can be convinced that this is important for the use of Windows in education.

**Reply by VirtualRusty** [reply]
Thanks for that detailed reply, that's very helpful. I've been terminating the apps via Task Manager in the exam room, and that works, but I'm the only one with those skills. Others will send the students back into the foyer where the IT desk is to get them to fix the issue (they are very busy at the start).

We load up students with so many requests via email and in the foyer that its unlikely we can add a request to quit open applications. The instances are low but sometimes we get a string of them in one exam.

I can't recall which brand of antivirus I terminated but I'll take note next year.

**Comment by marcgei** [comment]
Hi!
Following up on the above conversation, I have a question about prohibited processes:
as usual, starting the SEB while MS Teams ist running provokes a pop-up window saying "Teams needs to be closed" (or similar). By clicking "OK", the Teams Application is closed and SEB starts up (same behaviour for Zoom). But with TeamViewer, the pop-up which appears is a bit different: it says that the application needs to be closed and TeamViewer needs to be closed **manually** (Task Manager) and can only be startet after this. The SEB configuration is for both apps (Teams & TeamViewer) the same (both are on the prohibited processes list, "force quit" not ticked). Is this different behaviour during the start of SEB intentional? Or could we change anything in the configuration, that the behaviour for TeamViewer is the same as for Teams/Zoom? Thank you in advance!

**Reply by dbuechel** [reply]
The difference in the behavior is caused by the fact that, apparently, SEB is not able to terminate TeamViewer automatically and thus cannot allow the user to start a new session as long TeamViewer is running. Therefore, the user needs to terminate TeamViewer manually and then start SEB again.

**Reply by VirtualRusty** [reply]
OK that's useful information. I've just been working a 250 seat examination room again and some students run so many apps at boot up (startup items) its mind boggling - so I just give them an exam SOE exchange laptop. But what I saw consistently is dialogs on SEB telling students they must quit Steam and Discord before using SEB, but interestingly, if we just quit and restart SEB there is no issue, which is fine by me:-) But it would be better if SEB quit these during launch. I'm the only invigilator in the room with tech skills so at the start of exams I'm run off my feet dealing with SEB quirks like this.
